shared/settings/logout/index.tsx

```diff
@@ -15,133 +15,124 @@ export type Props = {
   waitingForResponse: boolean
 }
 
-type TestProps = {
-  checkPasswordIsCorrect: boolean | null
-  onCancel: () => void
-  onCheckPassword: (password: string) => void
-  onLogout: () => void
-}
-
 type State = {
   password: string
   showTyping: boolean
 }
 
-class OfferToCheckPassword extends React.Component<TestProps, State> {
+const HoverBox = Styles.isMobile
+  ? Kb.ClickableBox
+  : Styles.styled(Kb.ClickableBox)({
+      ':hover .text': {textDecoration: 'underline'},
+    })
+
+class LogOut extends React.Component<Props, State> {
   state = {
     password: '',
     showTyping: false,
   }
 
+  componentDidMount() {
+    this.props.onBootstrap()
+  }
+
   render() {
-    const inputType = this.state.showTyping ? 'passwordVisible' : 'password'
-    return (
-      <>
-        <Kb.Box2 direction="vertical" centerChildren={true} style={styles.offer}>
-          {!this.props.checkPasswordIsCorrect && (
-            <Kb.Box2 direction="vertical" centerChildren={true}>
-              <Kb.Input
-                errorText={
-                  this.props.checkPasswordIsCorrect === false ? 'Wrong password. Please try again.' : ''
-                }
-                hintText="Your password"
-                type={inputType}
-                value={this.state.password}
-                onChangeText={password => this.setState({password})}
-                uncontrolled={false}
-                style={styles.input}
-              />
-              <Kb.Checkbox
-                label="Show typing"
-                onCheck={showTyping => this.setState(prevState => ({showTyping: !prevState.showTyping}))}
-                checked={this.state.showTyping}
-              />
-            </Kb.Box2>
-          )}
-        </Kb.Box2>
-        {!this.props.checkPasswordIsCorrect ? (
-          <Kb.Box2 direction="vertical">
-            <Kb.ButtonBar align="center" direction="row" fullWidth={true}>
-              {!Styles.isMobile && (
-                <Kb.Button onClick={this.props.onCancel} label="Cancel" fullWidth={true} type="Dim" />
-              )}
+    const inputType = this.state.showTyping ? 'text' : 'password'
+    const keyboardType = this.state.showTyping && Styles.isAndroid ? 'visible-password' : 'default'
+    return this.props.hasRandomPW == null ? (
```

Comment:
1. don't ever use `==`
1. just check the falseyness of it. `!this.props.hasRandomPW ? `

Reply:
This was just copied over. I wasn't sure if the `==` was the original author's intent. I'll fix it.